react/features/toolbar/actions.web.js

```diff
@@ -0,0 +1,291 @@
+/* @flow */
+
+import Recording from '../../../modules/UI/recording/Recording';
+import SideContainerToggler
+    from '../../../modules/UI/side_pannels/SideContainerToggler';
+import UIEvents from '../../../service/UI/UIEvents';
+import UIUtil from '../../../modules/UI/util/UIUtil';
+
+import {
+    clearToolbarTimeout,
+    setAlwaysVisibleToolbar,
+    setSubjectSlideIn,
+    setToolbarButton,
+    setToolbarTimeout,
+    setToolbarTimeoutNumber,
+    setToolbarVisibility,
+    toggleToolbarButton
+} from './actions.native';
+
+export * from './actions.native';
+
+declare var $: Function;
+declare var APP: Object;
+declare var config: Object;
+declare var interfaceConfig: Object;
+
+/**
+ * Checks whether desktop sharing is enabled and whether
+ * we have params to start automatically sharing.
+ *
+ * @returns {Function}
+ */
+export function checkAutoEnableDesktopSharing(): Function {
+    return () => {
+        // XXX Should use dispatcher to toggle screensharing but screensharing
+        // hasn't been React-ified yet.
+
+        if (UIUtil.isButtonEnabled('desktop')
+                && config.autoEnableDesktopSharing) {
+            APP.UI.eventEmitter.emit(UIEvents.TOGGLE_SCREENSHARING);
+        }
+    };
+}
+
+/**
+ * Docks/undocks toolbar based on its parameter.
+ *
+ * @param {boolean} isDock - True if dock, false otherwise.
+ * @returns {Function}
+ */
+export function dockToolbar(isDock: boolean): Function {
+    return (dispatch: Dispatch<*>, getState: Function) => {
+        if (interfaceConfig.filmStripOnly) {
+            return;
+        }
+
+        const state = getState();
+        const { toolbarTimeout, visible } = state['features/toolbar'];
+
+        if (isDock) {
+            // First make sure the toolbar is shown.
+            visible || dispatch(showToolbar());
+
+            dispatch(clearToolbarTimeout());
+        } else if (visible) {
+            dispatch(setToolbarTimeout(() => {
+                dispatch(hideToolbar());
+            }, toolbarTimeout));
```

Comment:
@BeatC, the above is not as simple as possible. Please make an effort in the future to follow my lead. The above is to be written simply as follows: `() => dispatch(hideToolbar())`